Templating/TemplateGuesser.php

```diff
@@ -30,22 +30,33 @@ class TemplateGuesser
     protected $kernel;
 
     /**
+     * @var string[]
+     */
+    private $controllerPatterns;
+
+    /**
      * Constructor.
      *
-     * @param KernelInterface $kernel A KernelInterface instance
+     * @param KernelInterface $kernel             A KernelInterface instance
+     * @param string[]        $controllerPatterns Regexps extracting the controller name from its FQN.
      */
-    public function __construct(KernelInterface $kernel)
+    public function __construct(KernelInterface $kernel, array $controllerPatterns = null)
     {
+        if (null === $controllerPatterns) {
+            $controllerPatterns = array('/Controller\\\(.+)Controller$/');
```

Comment:
I don't see the downside of keeping the "default" and DX-wise, it's a much better experience.